ludwig/utils/upload_utils.py

```diff
@@ -205,3 +248,143 @@ def upload(
             return True
 
         return False
+
+
+class Predibase(BaseModelUpload):
+    def __init__(self):
+        self.pc = None
```

Comment:
Could we login at object construction time? That way, instead of needing to do:

```
predibase = Predibase()
predibase.login()
predibase.upload()
```

The caller can just do:

```
predibase = Predibase()
predibase.upload()
```

Reply:
@justinxzhao I agree, that's a cleaner/experience. I do think that most users won't use these classes directly, but instead use it via the LudwigModel API wrapper (https://github.com/ludwig-ai/ludwig/blob/master/ludwig/api.py#L1710), so it may be ok since it feels like an implementation detail. In either case, we can make this change!

Reply:
Addressed this by adding `self.login()` to `__init__` methods